src/main/java/io/airlift/compress/zstd/ZstdCompressor.java

```diff
@@ -97,9 +99,14 @@ else if (output.hasArray()) {
                         outputBase,
                         outputAddress,
                         outputLimit,
-                        CompressionParameters.DEFAULT_COMPRESSION_LEVEL);
+                        compressionLevel);
                 output.position(output.position() + written);
             }
         }
     }
+
+    public void setCompressionLevel(int compressionLevel)
```

Comment:
I don't think we should make this class have mutable state. We could make the field `final` make it a constructor argument. We'd want to preserve the existing no-argument constructor for simplicity and compatibility.

@martint thoughts?

Reply:
Why not have an overload of `compress` that accepts this input? I feel like I'm missing something.

Reply:
@DinoCassowary  Only compression level 3 and 4 via `Strategy.DFast` are supported. Not sure if setting the compression level is even necessary at the moment.